packages/toolpad-app/src/server/localMode.ts

```diff
@@ -964,6 +966,29 @@ export function getRequiredEnvVars(dom: appDom.AppDom): Set<string> {
   return new Set(allVars);
 }
 
+interface PaidFeature {
+  id: string;
+  label: string;
+}
+
+function detectPaidFeatures(application: Application): PaidFeature[] | null {
+  if (!application.spec || !application.spec.authorization) {
+    return null;
+  }
+
+  const hasRoles = Boolean(application?.spec?.authorization?.roles);
+  const hasAzureActiveDirectory = application?.spec?.authentication?.providers?.some(
+    (elems) => elems.provider === 'azure-ad',
+  );
+  const paidFeatures = [
+    hasRoles ? { id: 'roles', label: 'Role based access control' } : undefined,
+    hasAzureActiveDirectory
+      ? { id: 'azure-ad', label: 'Azure Active Directory authentication' }
```

Comment:
```suggestion
      ? { id: 'azure-ad', label: 'Azure Active Directory authentication provider' }
```

I would add the word `provider`, maybe "Azure AD authentication provider" not to be too long.

Reply:
Or maybe it's fine, whatever you prefer :D